src/odemis/gui/cont/features.py

```diff
@@ -0,0 +1,193 @@
+                                                                     events=wx.EVT_TEXT_ENTER,
+                                                                     va_2_ctrl=self._on_feature_name,
+                                                                     ctrl_2_va=self._on_cmb_feature_name_change,)
+
+        self._feature_status_va_connector = VigilantAttributeConnector(feature.status,
+                                                                       self._panel.cmb_feature_status,
+                                                                       events=wx.EVT_COMBOBOX,
+                                                                       ctrl_2_va=self._on_cmb_feature_status_change,
+                                                                       va_2_ctrl=self._on_feature_status)
+
+        self._feature_z_va_connector = VigilantAttributeConnector(feature.pos,
+                                                                  self._panel.ctrl_feature_z,
+                                                                  events=wx.EVT_TEXT_ENTER,
+                                                                  ctrl_2_va=self._on_ctrl_feature_z_change,
+                                                                  va_2_ctrl=self._on_feature_pos)
+
+    def _on_feature_pos(self, feature_pos):
+        # Set the feature Z ctrl with the 3rd (focus) element of the feature position
+        self._panel.ctrl_feature_z.SetValue(feature_pos[2])
+        save_features(self._tab.conf.pj_last_path, self._tab_data_model.main.features)
+
+    def _on_feature_name(self, _):
+        save_features(self._tab.conf.pj_last_path, self._tab_data_model.main.features)
+
+    def _on_cmb_feature_name_change(self, ):
+        feature = self._tab_data_model.main.currentFeature.value
+        value= self._panel.cmb_features.GetValue()
+        for stream in feature.streams.value:
+            stream.name.value = stream.name.value.replace(feature.name.value, value)
+        return value
+
+    def _on_feature_status(self, feature_status):
+        """
+        Update the feature status dropdown with the feature status
+        :param feature_status: (string) the updated feature status
+        """
+        self._panel.cmb_feature_status.SetValue(feature_status)
+        save_features(self._tab.conf.pj_last_path, self._tab_data_model.main.features)
+
+    def _on_cmb_features_change(self, evt):
+        """
+        Change the current feature based on the feature dropdown selection
+        """
+        index = self._panel.cmb_features.GetSelection()
+        if index == -1:
+            logging.warning("cmb_features selection = -1.")
+            return
+        selected_feature = self._panel.cmb_features.GetClientData(index)
+        self._tab_data_model.main.currentFeature.value = selected_feature
+
+    def _on_cmb_feature_status_change(self):
```

Comment:
Shouldn't it need an `evt` argument?

Reply:
Indeed but I removed it because` ctrl_2_va()` is called without passing the argument `evt`, didn't want to change it there in case it breaks somewhere else.
https://github.com/delmic/odemis/blob/44afe722ebba577bf2693a76a15eefeb17f2faab/src/odemis/gui/util/widgets.py#L101